### ISSUE TITLE
The conversion overhead here is not negligible and it will slow down training. How about moving this to the data pipeline?

### ISSUE THREAD

**Issue by RangiLyu** [open-mmlab/mmdetection#9508]
The conversion overhead here is not negligible and it will slow down training. How about moving this to the data pipeline?

_Originally posted by @tascj in https://github.com/open-mmlab/mmdetection/pull/9494#discussion_r1051396341_

**Comment by tascj** [comment]
Great. I tried `mask.to_tesnor` in `PackDetInputs` and it saved me 0.1x s/iter at batch size 32.

There's one related problem. You are using `ndarray` to store masks in BitmapMasks, which could be slow (long `data_time`) when used together with DataLoader. `torch.Tensor` with DataLoader does not have that problem.
You can speed up Mask R-CNN training a bit if using `torch.Tensor` in BitmapMasks. It would be nice to have that in mmdet 3.x.
